pkg/roachprod/vm/gce/create_sdk.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package gce
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "cloud.google.com/go/compute/apiv1/computepb"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/vm"
+	"github.com/cockroachdb/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+// buildInstanceProperties creates a computepb.InstanceProperties struct for the
+// GCP Compute SDK. This is the SDK equivalent of computeInstanceArgs()
+func (p *Provider) buildInstanceProperties(
+	l *logger.Logger,
+	opts vm.CreateOpts,
+	providerOpts *ProviderOpts,
+	startupScriptContent string,
+	zone string,
+) (*computepb.InstanceProperties, error) {
+	// Determine which image to use.
+	image := providerOpts.Image
+	imageProject := defaultImageProject
+
+	if providerOpts.useArmAMI() {
```

Comment:
Hmm, doesn't feel good duplicating all this logic between the bulk and non bulk API, but maybe its temporarily okay if we can eventually get rid of the non bulk API?

Reply:
Just chiming in to mention that it is temporary.
I have a [draft commit](https://github.com/cockroachdb/cockroach/pull/159084/changes/73dd1a9e72199d678b2afea797634879042f4012#diff-3339734a90a95640ef9141b4f19fb89c2dcb59e20994cf7f47c913711d2f7a6e) that completely drops `gcloud` to exclusively use the SDK.